src/database.providers.ts

```diff
@@ -1,9 +1,15 @@
 import { dbCredentialsString } from '@db/config';
+import * as usersSchema from '@users/users.model';
 import { type PostgresJsDatabase, drizzle } from 'drizzle-orm/postgres-js';
 import postgres from 'postgres';
 
 export const migrationsClient = postgres(dbCredentialsString, { max: 1 });
 
 export const queryClient = postgres(dbCredentialsString);
 
-export const db: PostgresJsDatabase = drizzle(queryClient);
+export const CombinedSchemas = { ...usersSchema };
+export type DatabaseSchema = PostgresJsDatabase<typeof CombinedSchemas>;
+export const db: DatabaseSchema = drizzle(queryClient, {
+  schema: CombinedSchemas,
+  logger: true,
+});
```

Comment:
```suggestion
export const db = drizzle(queryClient, {
  schema: { ...usersSchema },
  logger: true,
});
export type Database = typeof db;
```
A bit more elegant, also I'm a little uncomfortable with this line:

```ts
export type DatabaseSchema = PostgresJsDatabase<typeof CombinedSchemas>;
``` 

We shouldn't call it `DatabaseSchema` but `Database` - the `DatabaseSchema` is just the `CombinedSchemas` part

Reply:
Also I'm not sure if I'm reading [Schema in Multiple Files](https://orm.drizzle.team/docs/sql-schema-declaration#schema-in-multiple-files) correctly but this might work:
```suggestion
export const db = drizzle(queryClient, {
  schema: './src',
  logger: true,
});
export type Database = typeof db;
```
Worths a shot

Reply:
I think the docs you linked are for drizzle-kit, at least the section about schemas. The relevant drizzle-orm example can be found [here](https://orm.drizzle.team/docs/rqb)
<img width="842" alt="Screenshot 2025-03-16 at 0 44 06" src="https://github.com/user-attachments/assets/e2900ada-8398-498a-a635-ef4154895e26" />

Reply:
> A bit more elegant, also I'm a little uncomfortable with this line:
> 
> ```ts
> export type DatabaseSchema = PostgresJsDatabase<typeof CombinedSchemas>;
> ```
> 
> We shouldn't call it `DatabaseSchema` but `Database` - the `DatabaseSchema` is just the `CombinedSchemas` part

Thanks, the inferred type is definitely better